Adding support for HTML templates as external modules

sethx opened the pull request:
In reference to https://github.com/TypeStrong/grunt-ts/issues/208 .
This allows developers to define if the resulting TS will be an internal or an external module.

nycdotnet commented:
Hi @sethx - thanks for writing this and submitting it.  Looks cool.

Would you please make an update to the readme.md for this parameter on the options object, and also in the html2ts.md?

Also, I'm curious why you're generating both templates and only using one, rather than just generating the one you need.  Is there a reason?

If you're able to update this, I should be able to merge it in the next week or so.  Thanks.

sethx commented:
I've changed the code to function differently.
Instead of providing a way to generate an external module, i've decided to provide a way to specify a custom template. This also fixes the issue you pointed out with the generation function.
I've updated the documentation to reflect these changes.

nycdotnet commented:
Thanks - I'll take a look.

nycdotnet commented:
Thanks!  I think this looks good.
